core/src/test/java/jenkins/model/NewViewLinkTest.java

```diff
@@ -0,0 +1,46 @@
+package jenkins.model;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import hudson.model.Action;
+import hudson.model.View;
+import java.util.List;
+import org.junit.Test;
+
+public class NewViewLinkTest {
+
+    private NewViewLink newViewLink = new NewViewLink();
+
+    private View view = mock(View.class);
+
+    @Test
+    public void getActionsHasPermission() throws Exception {
+        when(view.hasPermission(any())).thenReturn(true);
+
+        final List<Action> actions = newViewLink.createFor(view);
+
+        assertEquals(1, actions.size());
+        final Action action = actions.get(0);
+        assertEquals(Messages.NewViewLink_NewView(), action.getDisplayName());
+        assertEquals(NewViewLink.ICON_FILE_NAME, action.getIconFileName());
+        assertEquals(NewViewLink.URL_NAME, action.getUrlName());
+    }
+
+    @Test
+    public void getActionsNoPermission() throws Exception {
+        when(view.hasPermission(any())).thenReturn(false);
```

Comment:
IMO hardly testing anything noteworthy beyond what you would get from code inspection of the class. What is of interest in practice is whether the link actually appears when you configure Jenkins with, say, a user with no `View.CREATE` permission, which is not immediately apparent (I had to dig through code to see how this would behave). More convincing would be `JenkinsRule.createDummySecurityRealm` + `MockAuthorizationStrategy` + `WebClient` etc.

In this case, it is maybe not worth testing at all, since a simple interactive test suffices to see what it really looks like, and there is no real harm from (say) displaying the link to someone without the permission—they would merely be directed to an error page.